src/utils/utils.rs

```diff
@@ -485,6 +486,19 @@ pub fn current_exe() -> Result<PathBuf> {
     env::current_exe().context(RustupError::LocatingWorkingDir)
 }
 
+pub fn current_name() -> Option<String> {
+    let arg0 = match process().var("RUSTUP_FORCE_ARG0") {
```

Comment:
I would put this on process() - `process().current_name()`

Reply:
Good point. That would work nicely with the currentprocess stuff, too. I'll try that tomorrow.

Reply:
I've pushed the updated version with this change. Intentionally not touching the error message for now.